crates/catalog-unity/src/credential.rs

```diff
@@ -415,39 +410,32 @@ impl TokenCredential for ImdsManagedIdentityOAuthProvider {
 #[cfg(test)]
 mod tests {
     use super::*;
-    use crate::data_catalog::client::mock_server::MockServer;
-    use futures::executor::block_on;
-    use hyper::body::to_bytes;
-    use hyper::{Body, Response};
-    use reqwest::{Client, Method};
+
+    // use hyper::{ Response};
+    use crate::client::retry::RetryConfig;
+    use httpmock::prelude::*;
+    use reqwest::Client;
     use tempfile::NamedTempFile;
 
     #[tokio::test]
     async fn test_managed_identity() {
-        let server = MockServer::new();
+        let server = MockServer::start_async().await;
 
         std::env::set_var(MSI_SECRET_ENV_KEY, "env-secret");
 
-        let endpoint = server.url();
-        let client = Client::new();
-        let retry_config = RetryConfig::default();
-
-        // Test IMDS
-        server.push_fn(|req| {
-            assert_eq!(req.uri().path(), "/metadata/identity/oauth2/token");
-            assert!(req.uri().query().unwrap().contains("client_id=client_id"));
-            assert_eq!(req.method(), &Method::GET);
-            let t = req
-                .headers()
-                .get("x-identity-header")
-                .unwrap()
-                .to_str()
-                .unwrap();
-            assert_eq!(t, "env-secret");
-            let t = req.headers().get("metadata").unwrap().to_str().unwrap();
-            assert_eq!(t, "true");
-            Response::new(Body::from(
-                r#"
+        let client = reqwest_middleware::ClientBuilder::new(Client::new()).build();
+
+        let _retry_config = RetryConfig::default();
```

Comment:
What's this for?

Reply:
I wasn't sure if I was going to remove it at that point from the testing, so it's a side effect left over from that.